net/respondd-module-airtime/src/respondd.c

```diff
@@ -22,8 +24,12 @@ static struct json_object *respondd_provider_statistics(void) {
 
 	ifaces = get_ifaces();
 	while (ifaces != NULL) {
-		get_airtime(wireless, ifaces->ifx);
-
+		ok = get_airtime(wireless, ifaces->ifx);
+		if (ok) {
+			newest_element_index = json_object_array_length(wireless)-1;
```

Comment:
Hmm, at a glance it seems that the return code of `get_airtime` is pretty useless - in some error paths it will return true even when no object has been added to the array. This can lead to `newest_element_index` pointing at a previous entry, or even worse, -1.

I think it would be better to change `get_airtime` to add the index instead of added the element afterwards.

Reply:
I think fixing the return code of get_airtime should be handled in a new issue;

There were three possible places where one could have added the phy info.
The one you propose does just replicate the return code of the wrapped `nl_send_dump`.
At that place one would have to add the info the same way one does in respond.c, but would have to add the wiphy int as parameter for `get_airtime`in some way.